test/router.js

```diff
@@ -461,14 +494,62 @@ test('should call `destroy` when exception is thrown inside async handler', t =>
   })
 })
 
+test('should call `destroy` when exception is thrown inside async handler (default route)', t => {
+  t.plan(2)
+  const fastify = Fastify()
+
+  t.tearDown(() => fastify.close())
+
+  fastify.register(fastifyWebsocket, {
+    handle: async function (conn, req) {
+      conn.on('error', err => {
+        t.equal(err.message, 'something wrong')
+        t.end()
+      })
+      throw new Error('something wrong')
+    }
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+    const ws = new WebSocket(
+      'ws://localhost:' + (fastify.server.address()).port + '/default-route'
+    )
+    const client = WebSocket.createWebSocketStream(ws, { encoding: 'utf8' })
+
+    client.on('error', (_) => { })
+    t.tearDown(client.destroy.bind(client))
+  })
+})
+
+test('should call default non websocket fastify route when no match is found', t => {
+  t.plan(2)
+  const fastify = Fastify()
+
+  t.tearDown(() => fastify.close())
+
+  fastify.register(fastifyWebsocket)
+  fastify.get('/ws', function handler (request, reply) {
+    reply.send({ hello: 'world' })
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+    get('http://localhost:' + (fastify.server.address()).port + '/wrong-route', function (response) {
+      t.equal(response.statusCode, 404)
+      t.end()
+    })
+  })
+})
+
```

Comment:
Would be good to add a test for both a websocket route and a non websocket route that are registered on the exact same path, and make sure both handlers can be invoked depending on the headers? I think that is a use case the HTTP spec supports. If it's too complicated to support we should probably warn or throw as well.

Reply:
Also would be good to add a test showing when the fastify server error handler is invoked. I don't think it's really possible to invoke it after the request has been fully upgraded, but, it might make sense for it to still be invoked before a request is hijacked. Same as the `preValidation` handlers or what have you, it'd be good if websocket-y requests had their errors handled the same way as other fastify routes before they actually become websocket-y.